faucet/src/tests.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2019 Alain Brenzikofer
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Unit tests for the encointer_balances module.
+
+use super::*;
+use crate::mock::{Balances, EncointerFaucet, EncointerReputationCommitments, System};
+use codec::Encode;
+use encointer_primitives::{
+	ceremonies::Reputation,
+	communities::{Degree, Location},
+	reputation_commitments::{DescriptorType, FromStr},
+};
+use frame_support::{assert_err, assert_ok};
+use mock::{master, new_test_ext, RuntimeOrigin, TestRuntime};
+use sp_runtime::DispatchError;
+use test_utils::{helpers::*, storage::*, *};
+
+#[test]
+fn purpose_id_is_stored_correclty() {
+	new_test_ext().execute_with(|| {
+		assert_eq!(EncointerFaucet::reputation_commitments_purpose_id(), 0,);
+		assert_eq!(
+			EncointerReputationCommitments::purposes(0),
+			DescriptorType::from_str("EncointerFaucet").unwrap(),
+		);
+	});
+}
+
+#[test]
+fn set_drip_amount_errs_with_bad_origin() {
+	new_test_ext().execute_with(|| {
+		assert_dispatch_err(
+			EncointerFaucet::set_drip_amount(
+				RuntimeOrigin::signed(AccountKeyring::Bob.into()),
+				10000u64,
+			),
+			DispatchError::BadOrigin,
+		);
+	});
+}
+
+#[test]
+fn set_drip_amount_works() {
+	new_test_ext().execute_with(|| {
+		System::set_block_number(System::block_number() + 1); // this is needed to assert events
+
+		assert_ok!(EncointerFaucet::set_drip_amount(RuntimeOrigin::signed(master()), 1001u64));
+
+		assert_eq!(EncointerFaucet::drip_amount(), 1001u64);
+		assert_ok!(EncointerFaucet::set_drip_amount(RuntimeOrigin::signed(master()), 1337u64));
+
+		assert_eq!(EncointerFaucet::drip_amount(), 1337u64);
+
+		assert_eq!(last_event::<TestRuntime>(), Some(Event::DripAmountUpdated(1337u64).into()));
+	});
+}
+
+#[test]
+fn minting_works() {
+	new_test_ext().execute_with(|| {
+		Balances::make_free_balance_be(&EncointerFaucet::account_id(), 101);
+		// free_balance - minimum_balance
+		assert_eq!(EncointerFaucet::pot(), 100);
+	});
+}
```

Comment:
misnomer: this test is not about minting, it tests if the `pot()` function returns the correct balance of the pot
and the test is brittle because it hardcodes the minimum_balance